### PR TITLE
Added Wiki

### PR DESCRIPTION
**Pull request by maybee16**
https://trello.com/c/Pm6tVx9a/60-imageservice-add-wiki

### CODE REVIEW

File: src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs [hunk 1]
```diff
@@ -63,14 +63,14 @@ public async Task<OperationResultResponse<CreateImageNewsResponse>> ExecuteAsync
...
         {
           Status = OperationResultStatusType.Failed,
           Errors = errors
         };
       }
 
-      OperationResultResponse<CreateImageNewsResponse> response = new();
+      OperationResultResponse<CreateImageWikiNewsResponse> response = new();
 
       DbImage dbImageNews = _mapper.Map(request);
```
Comment (vorobevamv) [inline src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs:75]: DbImage dbImage
or
DbImage dbImageWikiNews

File: src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs [hunk 2]
```diff
@@ -63,14 +63,14 @@ public async Task<OperationResultResponse<CreateImageNewsResponse>> ExecuteAsync
...
       DbImage dbPreviewNews = null;
```
Comment (vorobevamv) [inline src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs:76]: the same
DbImage dbPreview
or
DbImage dbPreviewWikiNews
and? please, fix all names

**Review by vorobevamv — APPROVED** [review]

File: src/ImageService.Data.Provider.MsSql.Ef/Migrations/20221006115000_InitialTable.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using LT.DigitalOffice.ImageService.Models.Dto.Constants;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace LT.DigitalOffice.ImageService.Data.Provider.MsSql.Ef.Migrations;
+
+[DbContext(typeof(ImageServiceDbContext))]
+[Migration("20221006115000_InitialTable")]
```
[the remark this answers is not in the document]
Comment (maybee16) [reply src/ImageService.Data.Provider.MsSql.Ef/Migrations/20221006115000_InitialTable.cs]: Done

File: src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs [hunk 4]
```diff
@@ -46,13 +46,13 @@ public class CreateImageNewsCommand : ICreateImageNewsCommand
       _validator = validator;
     }
 
-    public async Task<OperationResultResponse<CreateImageNewsResponse>> ExecuteAsync(CreateImageRequest request)
+    public async Task<OperationResultResponse<CreateImageWikiNewsResponse>> ExecuteAsync(CreateImageRequest request)
     {
       if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveNews))
```
[the remark this answers is not in the document]
Comment (maybee16) [reply src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs]: Done

File: src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs [hunk 5]
```diff
@@ -96,9 +96,16 @@ public async Task<OperationResultResponse<CreateImageNewsResponse>> ExecuteAsync
...
 
-      await _repository.CreateAsync(ImageSource.News, dbImagesNews);
+      if (request.Purpose == ImagePurpose.News)
+      {
+        await _repository.CreateAsync(ImageSource.News, dbImagesNews);
+      }
+      else
+      {
+        await _repository.CreateAsync(ImageSource.Wiki, dbImagesNews);
+      }
```
[the remark this answers is not in the document]
Comment (maybee16) [reply src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs]: Done

File: src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs (hunk 2 above)
Comment (maybee16) [reply src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs]: Done

File: src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs [hunk 6]
```diff
@@ -46,15 +47,15 @@ public class CreateImageNewsCommand : ICreateImageNewsCommand
       _validator = validator;
     }
 
-    public async Task<OperationResultResponse<CreateImageNewsResponse>> ExecuteAsync(CreateImageRequest request)
+    public async Task<OperationResultResponse<CreateImageWikiNewsResponse>> ExecuteAsync(CreateImageRequest request)
     {
-      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveNews))
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveNews)
+        || !await _accessValidator.HasRightsAsync(Rights.AddEditRemoveWiki))
```
Comment (vorobevamv) [inline src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs:53]: ImagePurpose = Wiki => check AddEditRemoveWiki right
ImagePurpose = News => check ddEditRemoveNews right

**Review by VLiamin — APPROVED** [review]

File: src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs (hunk 6 above)
Comment (maybee16) [reply src/ImageService.Business/Commands/ImageNews/CreateImageWikiNewsCommand.cs]: Done

**Pull request merged**